packages/graphql/src/translate/queryAST/factory/OperationFactory.ts

```diff
@@ -114,42 +141,102 @@ export class OperationsFactory {
         return this.createReadOperation(entity, resolveTree, context);
     }
 
-    // The current top-level Connection API is inconsistent with the rest of the API making the parsing more complex than it should be.
-    // This function temporary adjust some inconsistencies waiting for the new API.
-    // TODO: Remove it when the new API is ready.
-    private normalizeResolveTreeForTopLevelConnection(resolveTree: ResolveTree): ResolveTree {
-        const topLevelConnectionResolveTree = Object.assign({}, resolveTree);
-        // Move the sort arguments inside a "node" object.
-        if (topLevelConnectionResolveTree.args.sort) {
-            topLevelConnectionResolveTree.args.sort = (resolveTree.args.sort as any[]).map((sortField) => {
-                return { node: sortField };
-            });
-        }
-        // move the where arguments inside a "node" object.
-        if (topLevelConnectionResolveTree.args.where) {
-            topLevelConnectionResolveTree.args.where = { node: resolveTree.args.where };
-        }
-        return topLevelConnectionResolveTree;
-    }
-
-    private createCreateOperation(
-        entity: ConcreteEntityAdapter,
+    public createFulltextOperation(
+        entityOrRel: EntityAdapter | RelationshipAdapter,
```

Comment:
If it's available on top-level only maybe it's fair to assume/assert that it's always an ConcreteEntity, I may be wrong tho!